src/main/webapp/WEB-INF/templates/search.jsp

```diff
@@ -18,9 +18,12 @@
       <div class="card-container">
         <c:forEach items='${requestScope["cardList"]}' var="knowledgeCard">
           <div class="card">
-            <a href="${knowledgeCard.getLink()}" title="card-source"class="card-title">
-              ${knowledgeCard.getTitle()}
-            </a>
+            <span class="card-heading">
```

Comment:
We should usually avoid &lt;span&gt; unless there is no better semantic element. Here, &lt;header&gt; seems reasonable.